src/html-media-source.js

```diff
@@ -165,51 +179,57 @@ export default class HtmlMediaSource extends videojs.EventTarget {
     // active after a completed update (once it has/doesn't have videoTracks).
     // Once https://github.com/videojs/video.js/issues/2981 is resolved, switch to using
     // buffer.one instead of buffer.on.
-    //buffer.on('updateend', this.updateActiveSourceBuffers_.bind(this));
+    buffer.on('updateend', this.updateActiveSourceBuffers_.bind(this));
 
     this.sourceBuffers.push(buffer);
     return buffer;
   }
 
   updateActiveSourceBuffers_() {
-    let altAudioTrackEnabled;
-    let altAudioSourceBuffer;
+    // Retain the reference but empty the array
+    this.activeSourceBuffers_.length = 0;
 
+    let combined = 'enable';
+    let audioOnly = 'disable';
+
+    // TODO: find a better way to determine which sourcebuffers audio
+    // needs to be enabled this method relies on the track with kind 'main'
+    // being in the combined sourcebuffer. It is possible for the main track
+    // to have no audio and have a seprate track be the main audio
```

Comment:
I attempted to write up how to do this in a gist: https://gist.github.com/dmlap/26809ba67487ad413d47